moonraker/components/history.py

```diff
@@ -88,34 +89,58 @@ async def _handle_job_request(self, web_request):
 
     async def _handle_jobs_list(self, web_request):
         i = 0
-        end_num = len(self.history_ns)
+        count = 0
+        end_num = len(self.cached_job_ids)
         jobs = []
         start_num = 0
 
         before = web_request.get_float("before", -1)
         since = web_request.get_float("since", -1)
         limit = web_request.get_int("limit", 50)
         start = web_request.get_int("start", 0)
-        if start >= end_num or end_num == 0:
-            return {"count": 0, "jobs": []}
+        order = web_request.get_str("order", "desc")
 
-        for job_id in self.cached_job_ids:
-            job = self.history_ns[job_id]
-            if since != -1 and since > job.get('start_time'):
+        if not order in ["asc", "desc"]:
+            raise ServerError(f"Invalid `order` value: {order}")
```

Comment:
The other references in this module do
`return self.server.error()`
I think we should keep this consistent by changing the others or using self.server.error here.

Reply:
This is very good suggestion. Fixed.